lib/mobility/plugins/active_record/query.rb

```diff
@@ -22,6 +22,28 @@ def apply(attributes)
               extend FindByMethods.new(*attributes.names)
               singleton_class.send :alias_method, Mobility.query_method, :__mobility_query_scope__
             end
+            attributes.backend_class.include self
+          end
+
+          def attribute_alias(attribute)
+            "__mobility_%s__"  % attribute
+          end
+        end
+
+        # @note This is an ugly hack to return the value of a selected
+        #   Mobility attribute from the attribute accessor. We use the
+        #   "SELECT ... AS ..." form in the query, so ActiveRecord assigns
+        #   the returned value to the attributes hash, but in order to
+        #   actually return it we need to intercept the read here using
+        #   +instance_variable_get+. There should be a cleaner/safer way to
+        #   do this...
+        # TODO: Improve this.
+        def read(locale, **)
+          attr_set = model.instance_variable_get(:@attributes)
+          if attr_set && attr_set.key?(attr_alias = Query.attribute_alias(attribute))
```

Comment:
I've switched to using an alias when selecting, so `SELECT ... AS __mobility_title__`. This avoids conflicts with other stuff in ActiveRecord, but still passes grouping specs.

I still don't like the `instance_variable_get` here...